File: src/luogu/p5710/Main.cpp
// https://www.luogu.com.cn/problem/P5710

#include <cstdio>

bool a(int i) {
  return i % 2 == 0;
}

bool b(int i) {
  return i > 4 && i <= 12;
}

int main() {
  int i;
  scanf("%i", &i);
  printf("%i %i %i %i\n", a(i) && b(i) ? 1 : 0, a(i) || b(i) ? 1 : 0,
         (a(i) && !b(i)) || (!a(i) && b(i)) ? 1 : 0, !a(i) && !b(i) ? 1 : 0);
  return 0;
}

File: src/luogu/p5710/Main.py
# https://www.luogu.com.cn/problem/P5710
def a(i):
    return i % 2 == 0


def b(i):
    return 4 < i <= 12


i = int(input())
print('1' if a(i) and b(i) else '0', end=' ')
print('1' if a(i) or b(i) else '0', end=' ')
print('1' if (a(i) and not b(i)) or (not a(i) and b(i)) else '0', end=' ')
print('1' if not a(i) and not b(i) else '0')
